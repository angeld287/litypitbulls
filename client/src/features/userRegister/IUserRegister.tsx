import { IBase } from "../../interfaces/models/IBase"
import { IRegisterUser } from "../../interfaces/models/IUser"

export interface IUserRegisterSlice extends IBase {
    status: 'idle' | 'pending' | 'succeeded' | 'failed';
    user: IRegisterUser;
    isRegistering: boolean;
    isRegistered: boolean;
}