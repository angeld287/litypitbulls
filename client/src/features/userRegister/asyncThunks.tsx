import { createAsyncThunk } from "@reduxjs/toolkit";
import UserService from "../../apis/userService";
import IUserService from "../../interfaces/IUserService";
import { IRegisterUser } from "../../interfaces/models/IUser";

const userService: IUserService = new UserService();

export const registerAsync = createAsyncThunk(
    'userRegister/register',
    async (args: IRegisterUser) => {
        return await userService.register(args);
    }
);