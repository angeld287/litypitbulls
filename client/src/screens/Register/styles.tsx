const styles = {
    container: {
        width: 500,
        height: 400,
        margin: "50px auto",
        backgroundColor: "white",
        flexDirection: "column",
        justifyContent: "center",
        alignItems: "center",
    } as React.CSSProperties,
};

export default styles;