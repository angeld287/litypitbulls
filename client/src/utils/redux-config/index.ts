import { configureStore } from "@reduxjs/toolkit";
import userSessionReducer from '../../features/userSession/userSessionSlice';
import userRegisterReducer from '../../features/userRegister/userRegisterSlice';

export const store = configureStore({
    reducer: {
        userSession: userSessionReducer,
        userRegister: userRegisterReducer

    },
    middleware: (getDefaultMiddleware) => getDefaultMiddleware({
        serializableCheck: false
    })
});