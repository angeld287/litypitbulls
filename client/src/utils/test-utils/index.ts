// test-utils.js
import { render, queries, RenderOptions } from '@testing-library/react'
import { ReactElement } from 'react';
import * as customQueries from './custom-queries'
import * as functions from './functionalities';

const customRender = (
    ui: ReactElement,
    options?: Omit<RenderOptions, 'queries'>,
) => render(ui, { queries: { ...queries, ...customQueries }, ...options })


// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render, functions }