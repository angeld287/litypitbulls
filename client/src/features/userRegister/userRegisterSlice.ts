import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { RootState } from '../../app/store';
import { registerAsync } from './asyncThunks';
import { IUserRegisterSlice } from './IUserRegister';

export const initialState: IUserRegisterSlice = {
  user: {
    id: "",
    email: "",
    phoneNumber: "",
    password: "",
    confirmPassword: "",
    fullname: "",
    gender: "",
    userName: "",
  },
  status: 'idle',
  isRegistering: false,
  isRegistered: false,
};

export const userRegisterSlice = createSlice({
  name: 'userRegister',
  initialState,

  //Actions
  reducers: {
    setIsRegistering: (state, action: PayloadAction<boolean>) => {
      state.isRegistering = action.payload;
    },
  },

  //async operations
  extraReducers: (builder) => {

    builder
      .addCase(registerAsync.pending, (state) => {
        state.status = 'pending';
      })
      .addCase(registerAsync.fulfilled, (state, action) => {
        let data = action.payload.data;

        if (data.userId) {
          state.error = initialState.error;
          state.message = data.message;
          state.isRegistered = true;
        } else {
          state.isRegistered = false;
          state.isRegistering = true;
          state.error = data.errors ? data.errors : data
        }

        state.status = 'idle';
      })
      .addCase(registerAsync.rejected, (state) => {
        state.status = 'failed';
      })
  },
});

//Actions
export const { setIsRegistering } = userRegisterSlice.actions;

export const selectUserRegister = (state: RootState) => state.userRegister;

export default userRegisterSlice.reducer;
