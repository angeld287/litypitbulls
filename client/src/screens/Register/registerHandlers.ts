import { headers, json, _rest } from "../../utils/test-handlers-utils";

const register = _rest.post('http://localhost:3001/api/auth/register', (req, res, ctx) => {

    const _headers = headers(ctx);
    let _json = json(ctx, {});

    let _result: any = null;

    if (!req.body)
        return res(..._headers, _json, ctx.delay(100))

    _result = { errors: [{ message: "", value: "", param: "username", location: "body" }] }

    let _body: any = req.body; //existingadmin@test.com

    "Username cannot be blank."


    if (_body.username === "")
        _result.errors[0].message = "Username cannot be blank."

    if (_body.email === "")
        _result.errors[0].message = "E-mail cannot be blank."

    if (_body.email === "user1234")
        _result.errors[0].message = "E-mail is not valid."

    if (_body.phoneNumber === "")
        _result.errors[0].message = "Phone Number cannot be blank."

    if (_body.password === "")
        _result.errors[0].message = "Password cannot be blank."

    if (_body.password === "admin")
        _result.errors[0].message = "Password length must be atleast 8 characters."

    if (_body.confirmPassword === "")
        _result.errors[0].message = "Confirmation Password cannot be blank."

    if (_body.fullName === "")
        _result.errors[0].message = "fullName cannot be blank."

    if (_body.gender === "")
        _result.errors[0].message = "Gender cannot be blank."



    if (
        _body.email === "noexistingadmin@test.com"
        && _body.username === "noexistingadmin"
        && _body.phoneNumber === "8292619117"
        && _body.password === "admin2807"
        && _body.confirmPassword === "admin2807"
        && _body.fullName === "Test User Name"
        && _body.gender === "m"
    )
        _result = {
            userId: 10,
            message: "The user has been created successfully"
        };

    _json = json(ctx, {
        statusCode: "10000", message: "Success",
        data: _result
    });

    return res(..._headers, _json, ctx.delay(10))
});

const handlers = [register]

export default handlers