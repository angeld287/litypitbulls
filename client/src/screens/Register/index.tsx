import React, { useCallback, useEffect, useState } from "react"
import CustomForm from "../../components/CustomForm"
import { Content } from 'antd/lib/layout/layout';
import styles from "./styles";
//import { useAppDispatch } from "../../app/hooks";
import { MessageApi } from "antd/lib/message";
import { IValidationError } from "../../interfaces/models/IBase";
import { ICustomButton } from "../../components/CustomButton/ICustomButton";
import { selectUserRegister, setIsRegistering } from "../../features/userRegister/userRegisterSlice";
import { useAppDispatch, useAppSelector } from "../../app/hooks";
import { IRegisterUser } from "../../interfaces/models/IUser";
import { registerAsync } from "../../features/userRegister/asyncThunks";
import { ICustomFields } from "../../components/CustomForm/ICustomForm";


const Register: React.FC = () => {

    const [message, setMessage] = useState<MessageApi>()
    const [error, setError] = useState<IValidationError>()

    const register = useAppSelector(selectUserRegister);
    const dispatch = useAppDispatch()

    useEffect(() => {

        const _register = () => {

            setError({
                message: '',
                error: false
            });

            if (!message)
                return

            let registerError = register.error;
            if (registerError) {
                if (!Array.isArray(registerError)) {
                    setError({
                        message: registerError.message,
                        error: true
                    })
                    return message.error(registerError.message)
                }
                if (Array.isArray(registerError)) {
                    setError({
                        message: registerError[0].message,
                        error: true
                    })
                    return registerError.forEach(e => {
                        message.error(e.message)
                    });
                }
            }

            if (register.isRegistered) {
                message.success(register.message + " You can login.")
                dispatch(setIsRegistering(false))
            }
        }

        if (register.status === 'idle' && message) {
            _register()
        } else if (register.status === 'failed') {
            setError({
                message: 'Internal Error',
                error: true
            })
        }

    }, [register.status, register.error, message, register.message, register.isRegistered, dispatch]);

    const handleClicRegisterButton = useCallback(
        (user: IRegisterUser, message: MessageApi) => {
            setMessage(message);
            dispatch(registerAsync(user))
        }
        , [dispatch]
    );

    const closeRegisterScreen = useCallback(() => {
        dispatch(setIsRegistering(false));
    }, [dispatch])

    let inputFields: Array<ICustomFields> = [
        {
            name: 'email',
            input: {
                name: 'email',
                label: 'Email',
                defaultValue: '',
                disabled: false,
                type: 'email',
            }
        },
        {
            name: 'username',
            input: {
                name: 'username',
                label: 'Username',
                defaultValue: '',
                disabled: false,
                type: 'input',
            }
        },
        {
            name: 'phoneNumber',
            input: {
                name: 'phoneNumber',
                label: 'Phone Number',
                defaultValue: '',
                disabled: false,
                type: 'input',
            }
        },
        {
            name: 'password',
            input: {
                name: 'password',
                label: 'Password',
                defaultValue: '',
                disabled: false,
                type: 'password',
            }
        },
        {
            name: 'confirmPassword',
            input: {
                name: 'confirmPassword',
                label: 'Password Confirmation',
                defaultValue: '',
                disabled: false,
                type: 'password',
            }
        },
        {
            name: 'fullName',
            input: {
                name: 'fullName',
                label: 'Fullname',
                defaultValue: '',
                disabled: false,
                type: 'input'
            }
        },
        {
            name: 'gender',
            select: {
                getItemsNextToken: () => { },
                name: 'gender',
                placeholder: 'Select your Gender',
                defaultValue: '',
                disabled: false,
                label: 'Gender',
                items: [
                    {
                        id: 'm',
                        name: 'Male'
                    },
                    {
                        id: 'f',
                        name: 'Female'
                    }
                ],

            }
        }
    ]

    let btns: Array<ICustomButton> = [
        {
            color: 'blue',
            _key: 'login_btn',
            children: 'Login',
            htmlType: 'button',
            name: "login",
            onClick: closeRegisterScreen
        },
        {
            color: 'blue',
            _key: 'register_btn',
            children: 'Sign Up',
            loading: register.status === 'pending',
            htmlType: 'submit',
            name: "register",
        }
    ]

    return (
        <>
            <Content style={styles.container}>
                <CustomForm error={error} onSubmit={handleClicRegisterButton} fields={inputFields} buttons={btns} verticalButtons={false} loading={false} />
            </Content>
        </>
    )
}

export default Register