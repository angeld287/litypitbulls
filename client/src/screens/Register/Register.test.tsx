import { configureStore } from '@reduxjs/toolkit';
import { setupServer } from 'msw/lib/node';
import React from 'react';
import { Provider } from 'react-redux';
import { userRegisterSlice } from '../../features/userRegister/userRegisterSlice';
import { store } from '../../utils/redux-config';
import { render, RenderResult, functions, fireEvent, screen, waitFor, act } from '../../utils/test-utils';
import Register from './index'
import handlers from './registerHandlers';

let component: RenderResult;

const server = setupServer(...handlers)

describe("Login Test Suite", () => {

    beforeAll(() => server.listen({
        onUnhandledRequest: ({ headers, method, url }) => {
            if (headers.get("User-Agent") !== "supertest") {
                throw new Error(`Unhandled ${method} request to ${url}`);
            }
        },
    }));

    beforeEach(() => {
        if (component)
            component.unmount()

        act(() => {
            component = render(
                <Provider store={store}>
                    <Register />
                </Provider>
            )
        });

        server.resetHandlers()
    });

    afterAll(() => server.close())

    test('It must render the component correcly', () => {
        expect(component.asFragment()).toMatchSnapshot();
    })

    test('It must display basic register components', () => {
        expect(component.getByText(/Email/i)).toBeInTheDocument();
        expect(component.getByText(/Username/i)).toBeInTheDocument();
        expect(component.getByText(/Phone Number/i)).toBeInTheDocument();
        expect(component.getAllByText(/Password/i)[0]).toBeInTheDocument();
        expect(component.getByText(/Password Confirmation/i)).toBeInTheDocument();
        expect(component.getByText(/Fullname/i)).toBeInTheDocument();
        expect(component.getByText(/Select your Gender/i)).toBeInTheDocument();

        expect(component.getByText(/Sign Up/i)).toBeInTheDocument();
    });

    test('It must respond "E-mail cannot be blank." when email is blank.', async () => {

        functions.writeInInputFoundByPlaceHolder(null, /Type your Username/i, "noexistingadmin");
        functions.writeInInputFoundByPlaceHolder(null, /Type your Phone Number/i, "8292619117");
        functions.writeInInputFoundByPlaceHolder(null, /Type your Password/i, "admin2807");
        functions.writeInInputFoundByPlaceHolder(null, /Type your Password Confirmation/i, "admin2807");
        functions.writeInInputFoundByPlaceHolder(null, /Type your Fullname/i, "Test User Name");

        await waitFor(() => {
            functions.SelectItemInModalFoundByTestId("Gender", "select_id_gender", "Male");
        });

        await act(() => {
            fireEvent.click(screen.getByText(/Sign Up/i));
        });

        await waitFor(() => {
            expect(screen.getByText("E-mail cannot be blank.")).toBeInTheDocument();
        });
    });

    test('It should respond "Username cannot be blank." when Username is blank.', async () => {

        functions.writeInInputFoundByPlaceHolder(null, /Type your Email/i, "noexistingadmin@test.com");
        functions.writeInInputFoundByPlaceHolder(null, /Type your Phone Number/i, "8292619117");
        functions.writeInInputFoundByPlaceHolder(null, /Type your Password/i, "admin2807");
        functions.writeInInputFoundByPlaceHolder(null, /Type your Password Confirmation/i, "admin2807");
        functions.writeInInputFoundByPlaceHolder(null, /Type your Fullname/i, "Test User Name");

        await waitFor(() => {
            functions.SelectItemInModalFoundByTestId("Gender", "select_id_gender", "Male");
        });

        await act(() => {
            fireEvent.click(screen.getByText(/Sign Up/i));
        });

        await waitFor(() => {
            expect(screen.getByText("Username cannot be blank.")).toBeInTheDocument();
        });
    });

    test('It should respond "Phone Number cannot be blank." when PhoneNumber is blank.', async () => {

        functions.writeInInputFoundByPlaceHolder(null, /Type your Username/i, "noexistingadmin");
        functions.writeInInputFoundByPlaceHolder(null, /Type your Email/i, "noexistingadmin@test.com");
        functions.writeInInputFoundByPlaceHolder(null, /Type your Password/i, "admin2807");
        functions.writeInInputFoundByPlaceHolder(null, /Type your Password Confirmation/i, "admin2807");
        functions.writeInInputFoundByPlaceHolder(null, /Type your Fullname/i, "Test User Name");

        await waitFor(() => {
            functions.SelectItemInModalFoundByTestId("Gender", "select_id_gender", "Male");
        });

        await act(() => {
            fireEvent.click(screen.getByText(/Sign Up/i));
        });

        await waitFor(() => {
            expect(screen.getByText("Phone Number cannot be blank.")).toBeInTheDocument();
        });
    });

    test('It should respond "Password cannot be blank." when PhoneNumber is blank.', async () => {

        functions.writeInInputFoundByPlaceHolder(null, /Type your Username/i, "noexistingadmin");
        functions.writeInInputFoundByPlaceHolder(null, /Type your Email/i, "noexistingadmin@test.com");
        functions.writeInInputFoundByPlaceHolder(null, /Type your Phone Number/i, "8292619117");
        functions.writeInInputFoundByPlaceHolder(null, /Type your Password Confirmation/i, "admin2807");
        functions.writeInInputFoundByPlaceHolder(null, /Type your Fullname/i, "Test User Name");

        await waitFor(() => {
            functions.SelectItemInModalFoundByTestId("Gender", "select_id_gender", "Male");
        });

        await act(() => {
            fireEvent.click(screen.getByText(/Sign Up/i));
        });

        await waitFor(() => {
            expect(screen.getByText("Password cannot be blank.")).toBeInTheDocument();
        });
    });

    test('It should respond "Confirmation Password cannot be blank." when PhoneNumber is blank.', async () => {

        functions.writeInInputFoundByPlaceHolder(null, /Type your Username/i, "noexistingadmin");
        functions.writeInInputFoundByPlaceHolder(null, /Type your Email/i, "noexistingadmin@test.com");
        functions.writeInInputFoundByPlaceHolder(null, /Type your Phone Number/i, "8292619117");
        functions.writeInInputFoundByPlaceHolder(null, /Type your Password/i, "admin2807");
        functions.writeInInputFoundByPlaceHolder(null, /Type your Fullname/i, "Test User Name");

        await waitFor(() => {
            functions.SelectItemInModalFoundByTestId("Gender", "select_id_gender", "Male");
        });

        await act(() => {
            fireEvent.click(screen.getByText(/Sign Up/i));
        });

        await waitFor(() => {
            expect(screen.getByText("Confirmation Password cannot be blank.")).toBeInTheDocument();
        });
    });

    test('It should respond "fullName cannot be blank." when PhoneNumber is blank.', async () => {

        functions.writeInInputFoundByPlaceHolder(null, /Type your Username/i, "noexistingadmin");
        functions.writeInInputFoundByPlaceHolder(null, /Type your Email/i, "noexistingadmin@test.com");
        functions.writeInInputFoundByPlaceHolder(null, /Type your Phone Number/i, "8292619117");
        functions.writeInInputFoundByPlaceHolder(null, /Type your Password/i, "admin2807");
        functions.writeInInputFoundByPlaceHolder(null, /Type your Password Confirmation/i, "admin2807");

        await waitFor(() => {
            functions.SelectItemInModalFoundByTestId("Gender", "select_id_gender", "Male");
        });

        await act(() => {
            fireEvent.click(screen.getByText(/Sign Up/i));
        });

        await waitFor(() => {
            expect(screen.getByText("fullName cannot be blank.")).toBeInTheDocument();
        });
    });

    test('It should respond "Gender cannot be blank." when PhoneNumber is blank.', async () => {

        functions.writeInInputFoundByPlaceHolder(null, /Type your Username/i, "noexistingadmin");
        functions.writeInInputFoundByPlaceHolder(null, /Type your Email/i, "noexistingadmin@test.com");
        functions.writeInInputFoundByPlaceHolder(null, /Type your Phone Number/i, "8292619117");
        functions.writeInInputFoundByPlaceHolder(null, /Type your Password/i, "admin2807");
        functions.writeInInputFoundByPlaceHolder(null, /Type your Password Confirmation/i, "admin2807");
        functions.writeInInputFoundByPlaceHolder(null, /Type your Fullname/i, "Test User Name");

        await act(() => {
            fireEvent.click(screen.getByText(/Sign Up/i));
        });

        await waitFor(() => {
            expect(screen.getByText("Gender cannot be blank.")).toBeInTheDocument();
        });
    });
});