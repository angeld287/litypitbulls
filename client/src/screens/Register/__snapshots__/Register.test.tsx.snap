// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Test Suite It must render the component correcly 1`] = `
<DocumentFragment>
  <main
    class="ant-layout-content"
    style="width: 500px; height: 400px; margin: 50px auto; background-color: white; flex-direction: column; justify-content: center; align-items: center;"
  >
    <form
      autocomplete="off"
      class="ant-form ant-form-horizontal"
      id="basic"
    >
      <div>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-col-8 ant-form-item-label"
          >
            <label
              class=""
              for="basic_email"
              title="Email"
            >
              Email
            </label>
          </div>
          <div
            class="ant-col ant-col-16 ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  class="ant-input"
                  id="basic_email"
                  placeholder="Type your Email"
                  type="email"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-col-8 ant-form-item-label"
          >
            <label
              class=""
              for="basic_username"
              title="Username"
            >
              Username
            </label>
          </div>
          <div
            class="ant-col ant-col-16 ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  class="ant-input"
                  id="basic_username"
                  placeholder="Type your Username"
                  type="input"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-col-8 ant-form-item-label"
          >
            <label
              class=""
              for="basic_phoneNumber"
              title="Phone Number"
            >
              Phone Number
            </label>
          </div>
          <div
            class="ant-col ant-col-16 ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  class="ant-input"
                  id="basic_phoneNumber"
                  placeholder="Type your Phone Number"
                  type="input"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-col-8 ant-form-item-label"
          >
            <label
              class=""
              for="basic_password"
              title="Password"
            >
              Password
            </label>
          </div>
          <div
            class="ant-col ant-col-16 ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  class="ant-input"
                  id="basic_password"
                  placeholder="Type your Password"
                  type="password"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-col-8 ant-form-item-label"
          >
            <label
              class=""
              for="basic_confirmPassword"
              title="Password Confirmation"
            >
              Password Confirmation
            </label>
          </div>
          <div
            class="ant-col ant-col-16 ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  class="ant-input"
                  id="basic_confirmPassword"
                  placeholder="Type your Password Confirmation"
                  type="password"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-col-8 ant-form-item-label"
          >
            <label
              class=""
              for="basic_fullName"
              title="Fullname"
            >
              Fullname
            </label>
          </div>
          <div
            class="ant-col ant-col-16 ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  class="ant-input"
                  id="basic_fullName"
                  placeholder="Type your Fullname"
                  type="input"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-col-8 ant-form-item-label"
          >
            <label
              class=""
              for="basic_gender"
              title="Gender"
            >
              Gender
            </label>
          </div>
          <div
            class="ant-col ant-col-16 ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <div
                  class="ant-select ant-select-in-form-item ant-select-single ant-select-show-arrow ant-select-show-search"
                  data-testid="select_id_gender"
                  style="width: 100%;"
                >
                  <div
                    class="ant-select-selector"
                  >
                    <span
                      class="ant-select-selection-search"
                    >
                      <input
                        aria-activedescendant="select_id_gender_list_0"
                        aria-autocomplete="list"
                        aria-controls="select_id_gender_list"
                        aria-haspopup="listbox"
                        aria-owns="select_id_gender_list"
                        autocomplete="off"
                        class="ant-select-selection-search-input"
                        id="select_id_gender"
                        role="combobox"
                        type="search"
                        value=""
                      />
                    </span>
                    <span
                      class="ant-select-selection-placeholder"
                    >
                      Select your Gender
                    </span>
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-arrow"
                    style="user-select: none;"
                    unselectable="on"
                  >
                    <span
                      aria-label="down"
                      class="anticon anticon-down ant-select-suffix"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="down"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        style="text-align: right;"
      >
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-col-16 ant-col-offset-8 ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <button
                  class="ant-btn ant-btn-default"
                  style="color: blue;"
                  type="button"
                >
                  <span>
                    Login
                  </span>
                </button>
                <button
                  class="ant-btn ant-btn-default"
                  style="color: blue;"
                  type="submit"
                >
                  <span>
                    Sign Up
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </main>
</DocumentFragment>
`;
